mono/metadata/class-internals.h

```diff
@@ -351,6 +351,10 @@ struct MonoVTable {
 
 	guint32     imt_collisions_bitmap;
 	MonoRuntimeGenericContext *runtime_generic_context;
+	/* interp virtual method table */
+	gpointer *interp_vtable;
+	/* interp interface method table */
+	gpointer *interp_itable;
```

Comment:
It would be nice to squash that into `vtable`, otherwise we have this constant overhead of `2 * sizeof (gpointer)` bytes _per_ VTable instantiation in the runtime. On the other hand, this will make the implementation more complicated.

@vargaz do you have thoughts on this?

Reply:
The interface table at least can be stored at negative offsets from interp_vtable, similarly to the way the JIT does it.

Reply:
@vargaz @lewurm Moved the interp table inside MonoVTable.